### ISSUE TITLE
Querying repeated fields: how?

### ISSUE THREAD

**Issue by pontaoski** [tree-sitter/tree-sitter#1960]
(fields that look something like this in the node-types.json)

```json
        "multiple": true,
        "required": false,
        "types": [
          {
            "type": "decorator",
            "named": true
          }
        ]
```

Grammars can specify a field of a repeated item; but it's not clear how to access the children from code. All I see in the API is `ts_node_child_by_field_name` which only returns a singular node instead of multiple.

**Comment by nickcoutsos** [comment]
I just ran into this as well. From the underlying source code it looks like the important detail to work around it is an internal field map structure that isn't exposed.

It does seem that the Cursor object _can_ tell us what field a node has, so I've done the following:

```javascript
const map = {}
const cursor = node.walk()
cursor.gotoFirstChild()

do {
  const isNamed = cursor.currentNode().isNamed()
  const field = cursor.currentFieldName()
  if (isNamed && field) {
    map[cursor.nodeId] = field
  }
} while (cursor.gotoNextSibling())
```

In my example I'm building a map of node id to a field name, if that doesn't make sense for you then you can collect the details into some other form.

**Comment by Marcono1234** [comment]
It seems this cursor approach is also what java-tree-sitter is using: https://github.com/tree-sitter/java-tree-sitter/blob/v0.23.0/src/main/java/io/github/treesitter/jtreesitter/Node.java#L281-L297

(might be helpful for other bindings as well for understanding the logic)

**Comment by amaanq** [comment]
Yeah, it's not implemented in the C library as it'd require allocations (and at that point, you should just roll your own implementation on top of the cursor), but higher level libraries have it:
- [Rust](https://github.com/tree-sitter/tree-sitter/blob/f941277a9d57cecc92db29e439f5c9fe10bedfe1/lib/binding_rust/lib.rs#L1832)
- [Wasm](https://github.com/tree-sitter/tree-sitter/blob/f941277a9d57cecc92db29e439f5c9fe10bedfe1/lib/binding_web/tree-sitter-web.d.ts#L107)
...etc
